include/blade/modules/guppi/writer.hh

```diff
@@ -0,0 +1,131 @@
+#ifndef BLADE_MODULES_GUPPI_WRITER_HH
+#define BLADE_MODULES_GUPPI_WRITER_HH
+
+#include <filesystem>
+#include <string>
+
+#include "blade/base.hh"
+#include "blade/module.hh"
+
+extern "C" {
+#include "guppiraw.h"
+}
+
+namespace Blade::Modules::Guppi {
+
+template<typename IT>
+class BLADE_API Writer : public Module {
+ public:
+    struct Config {
+        std::string filepathStem;
+
+        bool directio;
+
+        U64 numberOfAntennas;
+        U64 numberOfBeams;
+        U64 numberOfFrequencyChannels;
+        U64 totalNumberOfFrequencyChannels;
+        U64 numberOfTimeSamples;
+        U64 numberOfPolarizations;
+
+        U64 blockSize = 512;
+    };
+
+    struct Input {
+        Vector<Device::CPU, IT> buf;
+    };
+
+    struct Output {
+    };
+
+    explicit Writer(const Config& config);
+
+    constexpr const U64 write() const {
+        return guppiraw_write_block_batched(this->file_descriptor, &this->gr_header, this->input.buf.data(), 1, this->getNumberOfFrequencyChannelBatches());
+    }
+
+    constexpr Vector<Device::CPU, IT>& getInput(U64 channel_batch_index) {
+        // TODO cast the offset-vector correctly
+        // return this->input.buf.data() + channel_batch_index * getInputBatchSize();
```

Comment:
Some sort of VectorView (a Vector sub-class with a shared underlying data pointer) is required to complete the Guppi::Writer in the most effective manner:
- the Guppi::Writer collects each coarse-channel chunked process-output and writes it correctly to disk.
- in order to do this, the chunked process-output needs to be iteratively written to the Guppi::Writer input...

Reply:
@luigifcruz